driver.go

```diff
@@ -449,72 +455,101 @@ func (d *Driver) StartTask(cfg *drivers.TaskConfig) (*drivers.TaskHandle, *drive
 			return nil, nil, fmt.Errorf("Trying to map ports but no network interface is available")
 		}
 	} else {
-		publishedPorts := []string{}
+		publishedPorts := []api.PortMapping{}
 		network := cfg.Resources.NomadResources.Networks[0]
 		allPorts := []structs.Port{}
 		allPorts = append(allPorts, network.ReservedPorts...)
 		allPorts = append(allPorts, network.DynamicPorts...)
 
 		for _, port := range allPorts {
-			hostPort := port.Value
+			hostPort := uint16(port.Value)
 			// By default we will map the allocated port 1:1 to the container
-			containerPort := port.Value
+			containerPort := uint16(port.Value)
 
 			// If the user has mapped a port using port_map we'll change it here
 			if mapped, ok := driverConfig.PortMap[port.Label]; ok {
-				containerPort = mapped
+				containerPort = uint16(mapped)
 			}
 
 			// we map both udp and tcp ports
-			publishedPorts = append(publishedPorts, fmt.Sprintf("%s:%d:%d/tcp", network.IP, hostPort, containerPort))
-			publishedPorts = append(publishedPorts, fmt.Sprintf("%s:%d:%d/udp", network.IP, hostPort, containerPort))
+			publishedPorts = append(publishedPorts, api.PortMapping{
+				HostIP:        network.IP,
+				HostPort:      hostPort,
+				ContainerPort: containerPort,
+				Protocol:      "tcp",
+			})
+			publishedPorts = append(publishedPorts, api.PortMapping{
+				HostIP:        network.IP,
+				HostPort:      hostPort,
+				ContainerPort: containerPort,
+				Protocol:      "udp",
+			})
 		}
-
-		createOpts.Publish = &publishedPorts
+		createOpts.ContainerNetworkConfig.PortMappings = publishedPorts
 	}
+	// -------------------------------------------------------------------------------------------
```

Comment:
```suggestion
```

Reply:
config MemoryMB is MB and podman wants bytes, hence limit=MemoryMB*1024*1024 should do the right thing. It's working but there is room for optimizations, config parameter is a left over from podman api.